### PR TITLE
fix(typescript-estree): support override modifier for parameter property

### PR DESCRIPTION
**Pull request by sosukesuzuki**
ref: https://github.com/prettier/prettier/issues/11017

Support TS 4.3 `override` modifier for parameter properties.

[TS Playground Link](https://www.typescriptlang.org/play?#code/MYGwhgzhAEBC0G8BQ1XWAewHYQC4CcBXYXDfACgAd8NcBTEugE2nzrCexAE91sAzAJYBzAFyJoYcXnyCswgNzQARuKyEAtsrr5oAXwCUiFGj1IzSUJBgARaHQAe9LExjxkqTDgLFSFarQM9CzQGABuOrJMdKhsHFy8fFhCYhJS0DJyiipqmtq6hsZoGYSUOuReKQYKJmZmQA)

### CODE REVIEW

File: packages/typescript-estree/tests/ast-alignment/fixtures-to-test.ts [hunk 1]
```diff
@@ -386,6 +386,11 @@ tester.addFixturePatternConfig('typescript/basics', {
      * This is intentional; babel is not checking types
      */
     'catch-clause-with-invalid-annotation',
+    /**
+     * [BABEL ERRORED, BUT TS-ESTREE DID NOT]
+     * SyntaxError: Unexpected token, expected ","
```
Comment (sosukesuzuki) [inline packages/typescript-estree/tests/ast-alignment/fixtures-to-test.ts:391]: babel/parser cannot parse override modifier for parameter properties, I'll report and fix it.

**Comment by bradzacher** [comment]
I'm not sure about this. It currently is an explicit TS error, so I think that babel has it right.

This should probably be a syntax error, and something we shouldn't support explicitly.

I think this is another case where prettier could access the underlying TS ast to look for the existence of a bad modifier (until such time that we have support for syntax errors?)

**Comment by sosukesuzuki** [comment]
For example, in this case the override modifier of the parameter property is required. [TypeScript Playground Link](https://www.typescriptlang.org/play?noImplicitOverride=true&strict=true#code/MYGwhgzhAEBC0G8BQ1XQA4FcBGICWw02YAXgFzQB2mAttgKYBO0AvNAIwDcKawA9pQgAXRpmBC+jABRZcBaADM+fCsMZ5KAcwA0GHPkLFGFanSYBKRD1QBfJHaShIMACLR6ADyH1KAExjwyGh6coR8AG5M6r70qEYmtAzMbOzW0AD06dARUXgx0Iz0AI6YeIW+0AD8afyCImIS0rIGisqqIho6IS3E5FSJFlbBwRCY6ExSSny6ACwATOZpdnZAA)
Also in TypeScript complier, override modifier for parameter property is allowed explicitly. (ref: https://github.com/microsoft/TypeScript/pull/43831)

**Comment by bradzacher** [comment]
OH! I see.
https://github.com/microsoft/TypeScript/pull/43831 was released as part of the RC.

When I built the feature, I used the playground to figure out what was supported.
The playground doesn't ever use the RC version, so I was working against `4.3-beta`.
So when I tested the parameter property case, it was a TS error.

But that PR obviously makes it supported - and testing against 4.3.2 it does work without an error.
This is correct then.

File: packages/typescript-estree/tests/ast-alignment/fixtures-to-test.ts (hunk 1 above)
Comment (sosukesuzuki) [reply packages/typescript-estree/tests/ast-alignment/fixtures-to-test.ts:391]: https://github.com/babel/babel/issues/13427

File: packages/typescript-estree/src/convert.ts [hunk 2]
```diff
@@ -1585,6 +1585,8 @@ export class Converter {
               hasModifier(SyntaxKind.ReadonlyKeyword, node) || undefined,
             static: hasModifier(SyntaxKind.StaticKeyword, node) || undefined,
             export: hasModifier(SyntaxKind.ExportKeyword, node) || undefined,
+            override:
+              hasModifier(SyntaxKind.OverrideKeyword, node) || undefined,
```
Comment (bradzacher) [inline packages/typescript-estree/src/convert.ts:1589]: I do hate that this emits `"override": undefined` if there isn't an override - but the entire node is built like this so 🤷‍♂️  it'll do for now.

We can revisit the entire AST later to fix this sort of weirdness and standardise on just emitting `false`.

**Review by bradzacher — CHANGES_REQUESTED** [review]
LGTM - one suggestion:
could you please add a test for this case with JUST the `override` as this emits as a parameter property:

```ts
class SpecializedComponent extends SomeComponent {
  constructor(override param: string) {
      super();
  }
}
```
[repl](https://www.typescriptlang.org/play?#code/MYGwhgzhAEDKD2BbApgYSQB3gO2dgLtAN4BQ050GYATmIgFzQT7UCW2A5tALzQDkfANwkAviRKhIMWBmTBWYEKwBeyACbpEWXAWjIAHvjxrpSNJhx5CpcsBzNqAV2D541ABTwAbsmps1yJQ0dIwO7BwAlMRkFOQQjrIeEcLkYmJAA)

**Review by bradzacher — APPROVED** [review]
thanks for this!

**Pull request merged**